### ISSUE TITLE
Cannot run Objective-C code

### ISSUE THREAD

**Issue by iabudiab** [osxpeppermint/peppermint#45]
Objective-C code fails to run with the following error:

```
ld: warning: ignoring file ../Library/Application Support/Peppermint/Data/Temp/9547.tmp, file was built for unsupported file format ( 0x2F 0x2A 0x2A 0
x2A 0x2A 0x2A 0x2A 0x2A 0x2A 0x2A 0x2A 0x2A 0x2A 0x2A 0x2A 0x2A ) which is not the architecture being linked (x86_64): /Users/iska/Library/Application Support/
Peppermint/Data/Temp/9547.tmp
 
Undefined symbols for architecture x86_64:
  "_main", referenced from:
     implicit entry/start for main executable
ld: symbol(s) not found for architecture x86_64
 
clang: error: linker command failed with exit code 1 (use -v to see invocation)
 
bash: ../Library/Application Support/Peppermint/Data/Temp/a.out: No such file or directory
```

I thought it may be sandboxing-related, but this also happens with the latest trial version downloaded directly from the site

**Comment by osxpeppermint** [comment]
Are you using the App Store version?

If so, given the App Store and Sandbox restrictions, just e-mail me your
App Store receipt and I will gladly send you a Peppermint License for the
off-AppStore version. 100% free, of course.

Just let me know.

*Ioannis Zafeiropoulos* *(*a.k.a.* Dr.Kameleon)*
Programmer / Web Developer
CEO & Chief Software Architect

<http://insili.co.uk>

*  InSili.co <http://InSili.co.uk>*
Phone  : (+34)636152727
E-mail  : i <drkameleon@gmail.com>nfo@insili.co.uk
              drkameleon@gmail.com

**Comment by iabudiab** [comment]
Thanks for pointing out the Script that runs the Objective-C code. The file extension is the culprit. Try compiling a source file with a **.tmp** extension directly with clang in the shell and you'll get the ```Undefined symbols for architecture x86_64:  "_main"```

If I'm not mistaken, any extension not recognised as a source code is passed to the linker as an object file.

PS: I've got Peppermint via the Creatable bundle, does it also have the sandboxing restriction?

**Comment by iabudiab** [comment]
Update: Saving the file with a correct extension then running it works.

Same behaviour can be seen for C and C++ snippets when run without saving first.
The temp files under the ```Application Support/Peppermint/Data/Temp/``` directory must be renamed, either automatically or via a dialog asking the user something like: "Run as C, C++ or ObjC?" in order to assign the correct extension.

**Comment by osxpeppermint** [comment]
The issue has been fixed and will be published as part of the upcoming 1.5 beta6 (and in the final 1.5 release).
